### PR TITLE
(4.x.x) Fix sorting sequences of more than 8192 items with collations

### PR DESCRIPTION
**Pull request by adamretter**
Backport of https://github.com/eXist-db/exist/pull/2667

### CODE REVIEW

File: exist-core/src/main/java/org/exist/xquery/value/OrderedValueSequence.java [hunk 1]
```diff
@@ -328,59 +319,64 @@ public Entry(final Item item, final int position) throws XPathException {
...
-                } catch (final XPathException e) {
                 }
+                // if the sort keys are equal, we need to order by the original position in the result sequence
+                if (cmp == Constants.EQUAL) {
+                    cmp = (pos > other.pos ? Constants.SUPERIOR : (pos == other.pos ? Constants.EQUAL : Constants.INFERIOR));
+                }
+                return cmp;
+            } catch (RuntimeException e) {
+                e.printStackTrace();
+                throw e;
```
Comment (dizzzz) [inline exist-core/src/main/java/org/exist/xquery/value/OrderedValueSequence.java]: no reporting with logger?

Comment (adamretter) [reply exist-core/src/main/java/org/exist/xquery/value/OrderedValueSequence.java]: Oops! That wasn't meant to make it into the PR. I have removed it now.

**Comment by joewiz** [comment]
Strange, appveyor failed with Java 11: https://ci.appveyor.com/project/AdamRetter/exist/builds/24281964.

**Comment by adamretter** [comment]
@joewiz looks like a blip in an unrelated part of eXist-db :-/

**Pull request merged**